Dockerfile

```diff
@@ -1,78 +1,138 @@
-FROM python:3.7.1-stretch
+# Dockerfile contains multiple levels of images--
+# First, a base image containing dependencies shared by all other images:
+#   - apt dependencies
+#   - rust
+#   - ntl
+#   - pbc
+#   - charm
+#   - base pip dependencies (cython & setup.py)
+# 
+# Thereafter, it add ever increasing levels of dependencies-- 
+#   - Test requirements (including doc requirements)
+#   - Dev requirements (including aws)
+#
+# In order to build and push this to dockerhub, run:
+# docker build . --target test-image --tag dsluiuc/honeybadgermpc-docker-base
```

Comment:
OK I see here is where the comments are..... I guess the Dockerfile is a natural enough place, but it should also be in a readme and/or wiki and/or the developer guide documentation

Reply:
It could be nice to add these statements to a script, like `scripts/build-and-push-to-dockerhub.sh` that also includes this documentation

Reply:
Will document this more thoroughly tonight, let's focus on getting this merged first.